File: CSharp/Oomph.Data/UF09Test/UFs/v301/Int4/ABC327_D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Oomph.Data.UF09Lib.UFs.v301;

// bipartite matching, fake vertexes
namespace UF09Test.UFs.v301.Int4
{
	// Test: https://atcoder.jp/contests/abc327/tasks/abc327_d
	class ABC327_D
	{
		static int[] Read() => Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
		static (int, int) Read2() { var a = Read(); return (a[0], a[1]); }
		static void Main() => Console.WriteLine(Solve() ? "Yes" : "No");
		static bool Solve()
		{
			var (n, m) = Read2();
			var a = Read();
			var b = Read();

			var uf = new UnionFind(2 * n + 1);

			for (int j = 0; j < m; j++)
			{
				var (u, v) = (a[j], b[j]);
				uf.Union(u, v + n);
				uf.Union(u + n, v);
			}
			return Enumerable.Range(1, n).All(v => !uf.AreSame(v, v + n));
		}
	}
}

File: Python/oomph-data-ws/uf_309_test/ufs/v301/Static1/ABC327_D.py
# https://atcoder.jp/contests/abc327/tasks/abc327_d
# bipartite matching, fake vertexes

from .....uf_309_lib.ufs.int_omega.unionfind_301 import UnionFind

n, m = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

uf = UnionFind(2 * n + 1)

for j in range(m):
    u, v = a[j], b[j]
    uf.union(u, v + n)
    uf.union(u + n, v)

r = True
for v in range(1, n + 1):
    if uf.are_same(v, v + n):
        r = False
        break
print("Yes" if r else "No")
